### PR TITLE
Generate Visual Recognition V4

### PR DESCRIPTION
**Pull request by mediumTaj**
### Summary
This pull request adds Visual Recognition v4 support to the .NET SDK

### CODE REVIEW

File: src/IBM.Watson.VisualRecognition.v4/VisualRecognitionService.cs [hunk 1]
```diff
@@ -0,0 +1,996 @@
...
+        /// <param name="imageUrl">An array of URLs of image files (.jpg or .png).
+        /// - Include a maximum of 20 images in a request.
+        /// - Limit each image file to 10 MB.
+        /// - Minimum width and height is 30 pixels, but the service tends to perform better with images that are at
+        /// least 300 x 300 pixels. Maximum is 5400 pixels for either height or width.
+        ///
+        /// You can also include images with the **images_file** parameter. (optional)</param>
+        /// <param name="threshold">The minimum score a feature must have to be returned. (optional)</param>
+        /// <returns><see cref="AnalyzeResponse" />AnalyzeResponse</returns>
+        public DetailedResponse<AnalyzeResponse> Analyze(string collectionIds, string features, List<FileWithMetadata> imagesFile = null, List<string> imageUrl = null, float? threshold = null)
```
Comment (lpatino10) [inline src/IBM.Watson.VisualRecognition.v4/VisualRecognitionService.cs]: `collectionIds` and `features` should both be lists.

File: test/VisualRecognition.v4.IntegrationTests/Properties/AssemblyInfo.cs [hunk 2]
```diff
@@ -0,0 +1,19 @@
+﻿using System.Reflection;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+// General Information about an assembly is controlled through the following
+// set of attributes. Change these attribute values to modify the information
+// associated with an assembly.
+[assembly: AssemblyConfiguration("")]
+[assembly: AssemblyCompany("IBM")]
+[assembly: AssemblyProduct("IBM.Watson.VisualRecognition.v3.IntegrationTests")]
```
Comment (lpatino10) [inline test/VisualRecognition.v4.IntegrationTests/Properties/AssemblyInfo.cs]: This one's for v4

File: test/VisualRecognition.v4.UnitTests/VisualRecognitionUnitTests.cs [hunk 3]
```diff
@@ -0,0 +1,1491 @@
...
+
+            client.WithAuthentication(Arg.Any<string>(), Arg.Any<string>())
+                .Returns(client);
+
+            return client;
+        }
+        #endregion
+
+        #region Environments
+        //#region List Environments
```
Comment (lpatino10) [inline test/VisualRecognition.v4.UnitTests/VisualRecognitionUnitTests.cs]: Looks like everything under this has to do with Discovery haha

File: examples/IBM.Watson.Assistant.v2.Examples/IBM.Watson.Assistant.v2.Examples.csproj [hunk 4]
```diff
@@ -6,11 +6,10 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="IBM.Cloud.SDK.Core" Version="1.0.0-rc9" />
+    <PackageReference Include="IBM.Cloud.SDK.Core" Version="1.0.0-rc91" />
```
Comment (lpatino10) [inline examples/IBM.Watson.Assistant.v2.Examples/IBM.Watson.Assistant.v2.Examples.csproj]: This version number seems like it really jumped 😂

Comment (mediumTaj) [reply examples/IBM.Watson.Assistant.v2.Examples/IBM.Watson.Assistant.v2.Examples.csproj]: lol yeah nuget didn't like `rc10`... in hindsight i should have started with `rc01`

File: test/VisualRecognition.v4.UnitTests/VisualRecognitionUnitTests.cs (hunk 3 above)
Comment (mediumTaj) [reply test/VisualRecognition.v4.UnitTests/VisualRecognitionUnitTests.cs]: ugh... my unit tests got lost somewhere :(

Comment (lpatino10) [reply test/VisualRecognition.v4.UnitTests/VisualRecognitionUnitTests.cs]: Nooo 😭😭

**Comment by mediumTaj** [comment]
@lpatino10 I think I addressed all of the comments. Regarding unit tests, it looks like I might have misplaced them - possibly on my machine at home. I updated the tests for now and will look for them tonight. If they are not there I think update these as part of our unit test generation activities.

File: test/VisualRecognition.v4.UnitTests/VisualRecognitionUnitTests.cs (hunk 3 above)
Comment (mediumTaj) [reply test/VisualRecognition.v4.UnitTests/VisualRecognitionUnitTests.cs]: :( seriously... thats what i get for working on multiple machines. Ill try to find them. If not, ill include them in unit test generation!

**Comment by lpatino10** [comment]
Changes look good to me but there are some CI errors 😕  I'll re-check once those are passing

**Pull request merged**